pkg/cluster/kubernetes/resource/load.go

```diff
@@ -158,10 +158,18 @@ func ParseMultidoc(multidoc []byte, source string) (map[string]KubeManifest, err
 		// contained resources we are after.
 		if list, ok := obj.(*List); ok {
 			for _, item := range list.Items {
-				objs[item.ResourceID().String()] = item
+				id := item.ResourceID().String()
+				if _, ok := objs[id]; ok {
+					return nil, fmt.Errorf(`duplicate definition of '%s' (in %s)`, id, source)
+				}
```

Comment:
Can we somehow factor out the duplicate-check and message? maybe in a local function?

Reply:
It occurred to me too, but I didn't bother because it would increase the line count, and the sites are close together. Have I drunk too much Go-aid? Possibly.